src/mongo/db/commands/list_databases.cpp

```diff
@@ -104,7 +111,7 @@ class CmdListDatabases : public Command {
                 const DatabaseCatalogEntry* entry = db->getDatabaseCatalogEntry();
                 invariant(entry);
 
-                int64_t size = entry->sizeOnDisk(txn);
+                int64_t size = nameOnly ? 0 : entry->sizeOnDisk(txn);
```

Comment:
Rather than setting `size = 0` here, why not skip the entire block from line 103 to 119? Reporting `sizeOnDisk: 0` and `empty: true` would seem undesirable.